include/seqan3/std/charconv_detail.hpp

```diff
@@ -66,6 +66,41 @@
 #include <seqan3/core/concept/core_language.hpp>
 #include <seqan3/std/concepts>
 
+// -----------------------------------------------------------------------------
+// structs: to_chars_result, from_chars_result and chars_format
+// -----------------------------------------------------------------------------
+
+namespace std
```

Comment:
Now non-detail stuff is in a detail folder o.O 
I deliberately separated the "public API stuff I copied" from the "detail stuff to make public API work that I messed with"

Reply:
That's mainly why I assigned you.

The problem is if we don't do it this way, we get weird include order issues. Because, `charconv_detail.hpp` needs stuff from `charconv.hpp`, but on the other hand `charconv.hpp` needs `charconv_detail.hpp`.

If you have an idea how to keep the current structure with the separation of the public and private API, but without this dependency loop. I'm happy to change this. The documentation will still be public even if it defined within the `detail` file.

Reply:
Where do we get these weird issues? I thought the include order is clearly defined
```
// some code that is needed in detail.hpp
#include <detail.hpp>
 // some code that is dependent on detail.hpp
```
I thought this is the way includes work? The only thing is that detail.hpp cannot be independently included somewhere else, but it should not need to be as it is a detail header?

Reply:
You are right, but like the title says no exceptions anymore, even our detail code should be includeable in any order.

Reply:
:( but why? when did we decide on that?

Reply:
We resolved this today at a quick strategy meeting, right?

Reply:
Yes by merging the detail header into the non-detail header, right 🤔

Reply:
done